it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+        OracleProfile.Builder oracleProfileBuilder = OracleProfile.newBuilder();
+        PostgresqlProfile.Builder postgresqlProfileBuilder = PostgresqlProfile.newBuilder();
+
+        if (this.source == Source.MYSQL) {
+            mysqlProfileBuilder.setHostname(hostname).setUsername(username).setPassword(password).setPort(3306);
+            connectionProfileBuilder.setMysqlProfile(mysqlProfileBuilder);
+        } else if (this.source == Source.ORACLE) {
+            oracleProfileBuilder.setHostname(hostname).setUsername(username).setPassword(password).setPort(1521);
+            connectionProfileBuilder.setOracleProfile(oracleProfileBuilder);
+        } else if (this.source == Source.ORACLE) {
+            postgresqlProfileBuilder.setHostname(hostname).setUsername(username).setPassword(password)
+                    .setPort(5432).setDatabase(postgresDb);
+            connectionProfileBuilder.setPostgresqlProfile(postgresqlProfileBuilder);
+        }
+
+        ConnectionProfile connectionProfile = connectionProfileBuilder
+                .setDisplayName(connectionProfileId)
+                .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                .build();
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfile)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    @Override
+    public ConnectionProfile createDestinationConnectionProfile(String connectionProfileId)
+            throws ExecutionException, InterruptedException {
+        ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder()
+                .setDisplayName(connectionProfileId)
+                .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance());
+
+        if (this.destination == Destination.GCS) {
+            connectionProfileBuilder
+                    .setGcsProfile(GcsProfile.newBuilder().setBucket(this.gcsBucketName).setRootPath(this.gcsRootPath));
+        }
+        // TODO: BQ profile builder
+
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfileBuilder)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
```

Comment:
I think there should be separate `createGCSDestinationConnectionProfile` and `createBigQueryDestinationConnectionProfile` methods, and the enum removed entirely. These two connection profiles require different arguments (e.g. BQ needs dataset/tables, GCS needs path), and as I said in my comment about the constructor, I think those parameters should be passed to this method instead of being stored as instance variables. Both will return a `ConnectionProfile` object to be used when creating the stream, and that way the caller can decide which type they want directly which IMO helps with user-error since all of the builder parameters are optional and can get missed.

Reply:
done